src/main/ipc/updater.ts

```diff
@@ -0,0 +1,161 @@
+import { ipcMain } from "electron";
+import {
+  RepluggedIpcChannels,
+  UpdateCheckResultFailure,
+  UpdateCheckResultSuccess,
+  UpdateInstallResultFailure,
+  UpdateInstallResultSuccess,
+  UpdaterType,
+} from "../../types";
+import { Octokit } from "@octokit/rest";
+import { CONFIG_PATHS } from "../../util";
+import { readdir, writeFile } from "fs/promises";
+import fetch from "node-fetch";
+import { join } from "path";
+import { Hash, createHash } from "crypto";
+
+const octokit = new Octokit();
+
+async function getHashRecursive(path: string, hash: Hash): Promise<void> {
+  const files = await readdir(path, { withFileTypes: true });
+
+  for (const file of files) {
+    if (file.isDirectory()) {
+      await getHashRecursive(path, hash);
+    } else {
+      hash.update(file.name);
+    }
+  }
+}
+
+async function getFolderHash(path: string): Promise<string> {
+  const hash = createHash("sha256");
+  await getHashRecursive(path, hash);
+  return hash.digest("hex");
+}
+
+async function github(
+  identifier: string,
+): Promise<UpdateCheckResultSuccess | UpdateCheckResultFailure> {
+  const [owner, repo] = identifier.split("/");
+  if (!owner || !repo) {
+    return {
+      success: false,
+      error: "Invalid repo identifier",
+    };
+  }
+
+  let res;
+
+  try {
+    res = await octokit.rest.repos.getLatestRelease({
+      owner,
+      repo,
+    });
+  } catch (err) {
+    return {
+      success: false,
+      // @ts-expect-error err tbd
+      error: err,
+    };
+  }
+
+  const asset = res.data.assets.find((asset) => asset.name.endsWith(".asar"));
```

Comment:
Instead of looking for if a file has .asar it should look for the filename because ideally the filename will be plugin/theme id. Also this makes it so monorepos wont work like [mine](https://github.com/E-boi/replugged-plugins).